include/dlaf/eigensolver/tridiag_solver/rot.h

```diff
@@ -31,43 +37,80 @@ namespace wrapper {
 template <Device D, class T>
 void sendCol(comm::Communicator& comm, comm::IndexT_MPI rank_dest, comm::IndexT_MPI tag,
              const T* col_data, const SizeType n, MPI_Request* req) {
-  if constexpr (D == Device::CPU) {
-    DLAF_MPI_CHECK_ERROR(MPI_Isend(col_data, static_cast<int>(n), dlaf::comm::mpi_datatype<T>::type,
-                                   rank_dest, tag, comm, req));
-  }
-  else {
-    dlaf::internal::silenceUnusedWarningFor(comm, rank_dest, tag, col_data, n, req);
-    DLAF_STATIC_UNIMPLEMENTED(T);
-  }
+  static_assert(D == Device::CPU, "This function works just with CPU memory.");
+
+  DLAF_MPI_CHECK_ERROR(MPI_Isend(col_data, static_cast<int>(n), dlaf::comm::mpi_datatype<T>::type,
+                                 rank_dest, tag, comm, req));
 }
 
 template <Device D, class T>
 void recvCol(comm::Communicator& comm, comm::IndexT_MPI rank_dest, comm::IndexT_MPI tag, T* col_data,
              const SizeType n, MPI_Request* req) {
-  if constexpr (D == Device::CPU) {
-    DLAF_MPI_CHECK_ERROR(MPI_Irecv(col_data, static_cast<int>(n), dlaf::comm::mpi_datatype<T>::type,
-                                   rank_dest, tag, comm, req));
-  }
-  else {
-    dlaf::internal::silenceUnusedWarningFor(comm, rank_dest, tag, col_data, n, req);
-    DLAF_STATIC_UNIMPLEMENTED(T);
-  }
+  static_assert(D == Device::CPU, "This function works just with CPU memory.");
+
+  DLAF_MPI_CHECK_ERROR(MPI_Irecv(col_data, static_cast<int>(n), dlaf::comm::mpi_datatype<T>::type,
+                                 rank_dest, tag, comm, req));
 }
 
 template <Device D, class T, class CommSender>
 auto scheduleSendCol(CommSender&& comm, comm::IndexT_MPI dest, comm::IndexT_MPI tag, const T* col_data,
                      const SizeType n) {
-  return dlaf::internal::whenAllLift(std::forward<CommSender>(comm), dest, tag, col_data, n) |
-         dlaf::comm::internal::transformMPI(sendCol<D, T>);
+  namespace di = dlaf::internal;
+
+  if constexpr (D == Device::CPU) {
+    return di::whenAllLift(std::forward<CommSender>(comm), dest, tag, col_data, n) |
+           dlaf::comm::internal::transformMPI(sendCol<D, T>);
+  }
+  else if constexpr (D == Device::GPU) {
+    namespace ex = pika::execution::experimental;
+    using dlaf::matrix::internal::CopyBackend_v;
+    using pika::execution::thread_priority;
+
+    return ex::just(memory::MemoryView<T, Device::CPU>{n}) |
+           ex::let_value([comm = std::forward<CommSender>(comm), dest, tag, col_data,
+                          n](memory::MemoryView<T, Device::CPU>& mem_view) mutable {
+             auto copy =
+                 di::whenAllLift(mem_view(), col_data, to_sizet(n) * sizeof(T),
```

Comment:
I think this is ok:
```suggestion
                 ex::just(mem_view(), col_data, to_sizet(n) * sizeof(T),
```
None of the arguments look like senders to me. This very slightly reduces template instantiations (`just` is cheaper than `when_all`).